File: FrogPond/csharp/UnitTests/UnitTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using FrogPond;

namespace UnitTests
{
    [TestClass]
    public class UnitTests
    {
        [TestMethod]
        public void Test_Solution_ReturnsTheCorrectTiming()
        {
            var leafDrop = new[] {1, 3, 1, 4, 2, 5};
            var frogJumps = 3;
            var pondSize = 7;

            var c = new Solution();
            int result = c.solution(leafDrop, pondSize, frogJumps);
            Assert.AreEqual(3, result);
        }
    }
}

File: FrogPond/java/src/tests/UnitTests.java
package tests;

import static org.junit.Assert.*;

import org.junit.Test;

import frogpond.Solution;

public class UnitTests {

	@Test
	public void testSolution_ReturnsTheCorrectTiming() {
		int[] A = {1,3,1,4,2,5};
		int X = 7;
		int D = 3;
		
        int result = Solution.solution(A,X,D);

        assertEquals(3, result);
	}

}
